stats/opentelemetry/client_metrics.go

```diff
@@ -0,0 +1,267 @@
+	// no-op.
+	if csh.mo.MeterProvider == nil {
+		return
+	}
+
+	meter := csh.mo.MeterProvider.Meter("gRPC-Go")
+	if meter == nil {
+		return
+	}
+
+	setOfMetrics := csh.mo.Metrics.metrics
+
+	clientMetrics := clientMetrics{}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.started"]; ok {
+		asc, err := meter.Int64Counter("grpc.client.attempt.started", metric.WithUnit("attempt"), metric.WithDescription("The total number of RPC attempts started, including those that have not completed."))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.attempt.started\", will not record")
+		} else {
+			clientMetrics.attemptStarted = asc
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.duration"]; ok {
+		cad, err := meter.Float64Histogram("grpc.client.attempt.duration", metric.WithUnit("s"), metric.WithDescription("End-to-end time taken to complete an RPC attempt including the time it takes to pick a subchannel."), metric.WithExplicitBucketBoundaries(DefaultLatencyBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.attempt.started\", will not record")
+		} else {
+			clientMetrics.attemptDuration = cad
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.sent_total_compressed_message_size"]; ok {
+		cas, err := meter.Int64Histogram("grpc.client.attempt.sent_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) sent across all request messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.attempt.sent_total_compressed_message_size\", will not record")
+		} else {
+			clientMetrics.attemptSentTotalCompressedMessageSize = cas
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.rcvd_total_compressed_message_size"]; ok {
+		car, err := meter.Int64Histogram("grpc.client.attempt.rcvd_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) received across all response messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.rcvd.sent_total_compressed_message_size\", will not record")
+		} else {
+			clientMetrics.attemptRcvdTotalCompressedMessageSize = car
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.call.duration"]; ok {
+		ccs, err := meter.Float64Histogram("grpc.client.call.duration", metric.WithUnit("s"), metric.WithDescription("This metric aims to measure the end-to-end time the gRPC library takes to complete an RPC from the application’s perspective."), metric.WithExplicitBucketBoundaries(DefaultLatencyBounds...))
```

Comment:
Description - "Time taken by gRPC to complete an RPC from application's perspective"

Reply:
Done.